src/core/qgsvectorlayer.h

```diff
@@ -953,10 +953,18 @@ class CORE_EXPORT QgsVectorLayer : public QgsMapLayer
     virtual QString subsetString();
 
     /**
-     * Query the provider for features specified in request.
+     * Query the layer for features specified in request.
      */
     QgsFeatureIterator getFeatures( const QgsFeatureRequest& request = QgsFeatureRequest() );
 
+    /**
+     * Query the layer for features that match the given filter expressions
+     * @param filterExpression An expression that features on this layer must match to be returned
+     *                         by this iterator
+     * @return A feature iterator
+     */
+    QgsFeatureIterator getFeatures( const QString& filterExpression );
+
```

Comment:
Wouldn't it be easier to just remove the `explicit` from the `QgsFeatureRequest` ctors?